homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Sensor for displaying the number of result from Flume."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from pyflume import FlumeDeviceList, FlumeData
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+Logger = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Flume Sensor"
+
+CONF_CLIENT_ID = "client_id"
+CONF_CLIENT_SECRET = "client_secret"
+FLUME_TYPE_SENSOR = 2
+
+SCAN_INTERVAL = timedelta(minutes=1)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=45)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Flume sensor."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+    client_id = config[CONF_CLIENT_ID]
+    client_secret = config[CONF_CLIENT_SECRET]
+    time_zone = str(hass.config.time_zone)
+    name = config[CONF_NAME]
+    flume_entity_list = []
+
+    flume_devices = FlumeDeviceList(username, password, client_id, client_secret)
+
+    try:
+        for device in flume_devices.device_list:
+            if device["type"] == FLUME_TYPE_SENSOR:
+                flume = FlumeData(
+                    username,
+                    password,
+                    client_id,
+                    client_secret,
+                    device["id"],
+                    time_zone,
+                    SCAN_INTERVAL,
+                )
+                flume_entity_list.append(FlumeSensor(flume, f"{name} {device['id']}"))
+    except KeyError:
```

Comment:
Is there a chance that `"type"` key is missing in the device dict or just that the value of that key is not `FLUME_TYPE_SENSOR`?

Reply:
I have not seen an instance of the type key existing, but I do have an error to catch it. There is a chance that a `FLUME_TYPE_SENSOR` might not exist in the user's device list.

Reply:
Is there an action item you would like me to address for either of those occurring? @MartinHjelmare

Reply:
I'm wondering why we are catching a `KeyError`. What key might be missing?

Reply:
Either `id` or `type`. Although if the flume api returns data this shouldn't be the case. Originally I had a more broad exception and narrowed it down to only key errors occurring in order to meet the automated tests.